File: backend/txn-service/src/main/java/com/meiyun/txn/Appointment.java
package com.meiyun.txn;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.OffsetDateTime;

/**
 * 预约明细（M4-01）。status 四态：已预约/已到店/未到诊/已取消。
 * 月度权威口径仍是 appointment_month（ID-7：1286=1112+174）。
 */
@Entity
@Table(name = "appointment")
@Getter @Setter @NoArgsConstructor
public class Appointment {

    @Id
    @Column(name = "appt_no", length = 24)
    private String apptNo;

    @Column(name = "customer_id", length = 16)
    private String customerId;

    @Column(name = "store_code", length = 16)
    private String storeCode;

    @Column(nullable = false, length = 64)
    private String project;

    @Column(name = "appt_date", nullable = false)
    private LocalDate apptDate;

    @Column(name = "appt_time", nullable = false, length = 8)
    private String apptTime;

    @Column(length = 32)
    private String doctor;

    @Column(nullable = false, length = 16)
    private String source;                    // B端登记/C端小程序/C端App

    @Column(nullable = false, length = 8)
    private String status;                    // 已预约/已到店/未到诊/已取消

    @Column(name = "arrived_at")
    private OffsetDateTime arrivedAt;

    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt;

    @PrePersist
    void prePersist() {
        if (createdAt == null) createdAt = OffsetDateTime.now();
        if (source == null) source = "B端登记";
        if (status == null) status = "已预约";
    }
}

File: frontend/src/api/appointment.ts
// ============================================================
// Appointment 聚合 API（对接 txn-service 预约域，真实读写）
// 库内状态/来源均为中文枚举：状态 已预约/已到店/未到诊/已取消；
// 来源 B端登记/C端小程序/C端App。后端读模型冗余客户/门店/医生中文名。
// ============================================================
import client from './client'

export interface AppointmentView {
  apptNo: string
  customerId: string | null
  customerName: string | null
  storeCode: string | null
  storeName: string | null
  project: string
  apptDate: string
  apptTime: string
  doctor: string | null
  doctorName: string | null
  source: string
  status: string
  arrivedAt: string | null
  createdAt: string | null
}

export interface BoardStats {
  total: number
  booked: number
  arrived: number
  noShow: number
  cancelled: number
  arrivalRate: number
}

export interface CreateApptCmd {
  customerId: string
  storeCode: string
  project: string
  apptDate: string
  apptTime: string
  doctor?: string | null
  source?: string
  operator?: string
}

export const createAppointment = (cmd: CreateApptCmd) =>
  client.post<AppointmentView>('/txn/appointment', cmd)

export const rescheduleAppointment = (no: string, apptDate: string, apptTime: string) =>
  client.post<AppointmentView>(`/txn/appointment/${no}/reschedule`, { apptDate, apptTime })

export const checkIn = (no: string) =>
  client.post<AppointmentView>(`/txn/appointment/${no}/check-in`)

export const cancelAppointment = (no: string) =>
  client.post<AppointmentView>(`/txn/appointment/${no}/cancel`)

export const noShow = (no: string) =>
  client.post<AppointmentView>(`/txn/appointment/${no}/no-show`)

export const listAppointments = (storeCode?: string, date?: string) =>
  client.get<AppointmentView[]>('/txn/appointment', { params: { storeCode, date } })

export const appointmentBoard = (storeCode?: string, date?: string) =>
  client.get<BoardStats>('/txn/appointment/board', { params: { storeCode, date } })

/** 客户跨店预约/到店交叉校验（撞单辅助） */
export const crossCheck = (customerId: string) =>
  client.get<AppointmentView[]>(`/txn/appointment/cross-check/${customerId}`)
